cli/src/main/java/org/jboss/as/cli/CommandCompleter.java

```diff
@@ -139,6 +142,31 @@ protected int doComplete(CommandContext ctx, String buffer, int cursor, List<Str
                 candidates.remove(notIndex);
             }
         }
+
+        // https://issues.jboss.org/browse/WFCORE-1714
+        if (unresolvedVariableException == false) {
+            int index = buffer.lastIndexOf('$');
+            if (index != -1) {
+                String variable = buffer.substring(index + 1);
+                if (buffer.endsWith(variable)) {
+                    List<String> candidateVariables = new ArrayList<>();
+                    for (String var : ctx.getVariables()) {
+                        if (var.startsWith(variable)) {
+                            candidateVariables.add(var);
+                        }
+                    }
+                    if (candidateVariables.size() > 1) {
```

Comment:
You could filter out the existing variable first, then simply check if the list of variables isEmpty? It would seem more natural to me.